devito/types/dimension.py

```diff
@@ -676,22 +676,24 @@ class ConditionalDimension(DerivedDimension):
     is_NonlinearDerived = True
     is_Conditional = True
 
-    def __new__(cls, name, parent, factor=None, condition=None, indirect=False):
+    def __new__(cls, name, parent, factor=None, condition=None, indirect=False,
+                spacing=None):
         return ConditionalDimension.__xnew_cached_(cls, name, parent, factor,
-                                                   condition, indirect)
+                                                   condition, indirect, spacing)
 
-    def __new_stage2__(cls, name, parent, factor, condition, indirect):
+    def __new_stage2__(cls, name, parent, factor, condition, indirect, spacing):
         newobj = DerivedDimension.__xnew__(cls, name, parent)
         newobj._factor = factor
         newobj._condition = condition
         newobj._indirect = indirect
+        newobj._spacing = spacing or parent.spacing
         return newobj
 
     __xnew_cached_ = staticmethod(cacheit(__new_stage2__))
 
     @property
     def spacing(self):
-        return self.factor * self.parent.spacing
+        return self.factor * self._spacing
```

Comment:
much more clear if you return directly ``self._spacing`` here, and change ``__new_stage2__`` such that ``newobj._spacing = factor*(spacing or parent.spacing)``